### DISCUSSION TITLE
How to format a USER field with multiple users

### DISCUSSION THREAD

**Discussion by zkent** [microsoft-search/pnp-modern-search#3053] [Q&A]
I have a column that contains people and can have multiple people per record, separated by a `\n\n`.  I get a result like this 

`"FieldNameOWSUSER": "user.one@domain.org | Three, User | 693A30232E667C6D656D626572736869707C74686F6D61732E6D656C6C6F72406D6F7267616E6C657769732E636F6D i:0#.f|membership|user.one@domain.org\n\nuser.two@domain.org | Two, User | 693A30232E667C6D656D626572736869707C66656C6970652E616C696365406D6F7267616E6C657769732E636F6D i:0#.f|membership|user.two@domain.org\n\nuser.three@domain.org | Three, User | 693A30232E667C6D656D626572736869707C68756D626572746F2E706164696C6C61406D6F7267616E6C657769732E636F6D i:0#.f|membership|user.three@domain.org"`

**What I want to do**

Display the people similar to the way SharePoint does in the native list.  See below (I blurred the names).
![image](https://github.com/microsoft-search/pnp-modern-search/assets/549893/b74fc50a-8b54-4de9-8836-ddbccabc52b4)

I'd like replicate that badge-style with the native hover feature.

At my work, I don't have the appropriate access to create a custom handlebars helper.   I have been trying to split the `FieldNameOWSUSER` value using the `each` and `split` handlebars helpers but I am not having good luck getting split to work with `\n\n`

Is there something I am missing?  Is there an easier way?  

Thanks in advance.

**Comment by zkent** [comment]
Is there a way to return that column as nested SPSPeople content types instead of a string?  I am looking at the people.html template and that is what I want to recreate in my output.

**Reply by kasperbolarsen** [reply]
Interesting question 🤔  I will look into it using a multi value people site column as the base.
Is that the setup you are using?

**Reply by gretchunkim** [reply]
@zkent  Very curious if any one got this to work as the native web part for People isn't that space saving (too large, border,  multi-people can't be listed). You might want to check out [SharePoint Online list view ](https://github.com/pnp/List-Formatting/blob/master/view-samples/multi-person-displays/multi-person-displays.json) for multi-person for any hint.  I'm assuming you've already checked discussion [2756](https://github.com/microsoft-search/pnp-modern-search/discussions/2756 ).

**Reply by zkent** [reply]
@gretchunkim I have been trying out other views and checking the resulting code.  I'll test the list view next (if I haven't already).  

Regarding the helpers discussed in 2756, I have tried using `split` but it doesn't seem to work with new lines such as `\n\n`, which is the delimiter that I am getting with multi-person columns.  That issue is the reason I am here asking.

**Reply by wobba** [reply]
If someone can contribute a helper similar to `GetAttachments` we could add that to help on splitting multi-people fields.

**Comment by zkent** [comment]
>  ... multi value people site column as the base. Is that the setup you are using?

Yes, that is what I am using

**Comment by bashdx** [comment]
The same issue holds true to the "LinkOfficeChild" property. If you have multiple attachments to a list item, the engine will return a string which contains a "\n\n" as seperator for the entries.

**Reply by bashdx** [reply]
Edit: Found out about the "GetAttachments" helper.

https://microsoft-search.github.io/pnp-modern-search/extensibility/templating/
